app/Http/Controllers/MovieController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Movie;
+use App\Models\Review;
+use Illuminate\Http\Request;
+
+class MovieController extends Controller
+{
+    public function index()
+    {
+        $movies = Movie::all()->sortBy('created_at');
+        return view('movies.index', ['movies' => $movies]);
+    }
+
+    public function create()
+    {
+        //
+    }
+
+    public function store(Request $request)
+    {
+        //
+    }
+
+    public function show(Movie $movie)
+    {
+        $genres = Movie::with('genres')->get();
+        $castings = Movie::with('celebs')->get();
+        $reviews = Movie::with('reviews')->get();
+        $duration = explode('.', (string) round($movie->duration/60, 2));
```

Comment:
This can be refactored a bit, my friend:

In this scenario, you are already passing the movie in the parameter, so there is no need to use `Movie::with`. Also, with the `with` you can actually have more than one relationship, so for example `with(['reviews', 'genres', 'celebs'])`.

This would also just get the relevant information regarding the movie you are querying, instead of querying all the movies and getting all the reviews, then querying again to get all the celebs and then checking on the view to see if the review is the one you want. We usually want to avoid doing much logic in the view and do all the necessary things on the controller when we can. That way all the logic is on a single location :D 

In this case, you can update by using the `$movie` and join the relations.

```
$movie->with(['genres', 'celebs', 'reviews'])->get();
```

Then you can pass only the movie to the view, and all the information would be accessible. So in the view, if you want to add the movie's reviews you would only need to do `$movie->reviews`.

Reply:
Woow! That is really very simple! Thank you so much!